synfig-studio/src/gui/duckmatic.cpp

```diff
@@ -851,35 +846,64 @@ Duckmatic::snap_point_to_grid(const synfig::Point& x)const
 	Point ret(x);
 	float radius(0.1/zoom);
 
-	GuideList::const_iterator guide_x,guide_y;
-	bool has_guide_x(false), has_guide_y(false);
+	GuideList::const_iterator guide;
 
-	guide_x=find_guide_x(ret,radius);
-	if(guide_x!=guide_list_x_.end())
-		has_guide_x=true;
+	guide = find_guide(ret,radius);
 
-	guide_y=find_guide_y(ret,radius);
-	if(guide_y!=guide_list_y_.end())
-		has_guide_y=true;
+	bool has_guide_x_ = has_guide_x;
+	bool has_guide_y_ = has_guide_y;
 
 	if(get_grid_snap())
 	{
 		Point snap(
 			floor(ret[0]/get_grid_size()[0]+0.5)*get_grid_size()[0],
 			floor(ret[1]/get_grid_size()[1]+0.5)*get_grid_size()[1]);
 
-		if(std::fabs(snap[0]-ret[0])<=radius && (!has_guide_x || std::fabs(snap[0]-ret[0])<=std::fabs(*guide_x-ret[0])))
-			ret[0]=snap[0],has_guide_x=false;
-		if(std::fabs(snap[1]-ret[1])<=radius && (!has_guide_y || std::fabs(snap[1]-ret[1])<=std::fabs(*guide_y-ret[1])))
-			ret[1]=snap[1],has_guide_y=false;
+		if(std::fabs(snap[0]-ret[0])<=radius && (!has_guide_x || std::fabs(snap[0]-ret[0])<=std::fabs((guide->point)[0]-ret[0])))
```

Comment:
I believe you should check the distances now the guideline may have any angle:
What is nearest: the vertical grid line, the horizontal grid line or (if guideline snap is enabled) the nearest guideline?

In latter case, then you'll have to compute where is the nearest point in guideline.

Reply:
agreed like in the guide snap functionality, I guess I forgot to update this. done.